aapcs64/aapcs64.rst

```diff
@@ -898,6 +898,15 @@ registers, or if it is a function that returns results in such registers,
 it must ensure that p4-p15 are preserved across the call. In other cases
 it need not preserve any scalable predicate register contents.
 
+aarch64_sve_vector_pcs
+^^^^^^^^^^^^^^^^^^^^^^
+
+The attribute `aarch64_sve_vector_pcs` can be specified to ensure a subroutine
+preserves the contents of Scalable Vector Registers z8-z23 and Scalable 
+Predicate Registers p4-15. This attribute can be used when a subroutine can
+benefit from preserving scalable vector and scalable predicate registers but
+has no scalable vector or scalable predicate arguments or return results.
```

Comment:
HI @MDevereau, apologies for the delay in my feedback.

This to me is more clear than the original version, thank you.

I think there is a risk that the name `aarch64_sve_vector_pcs` will get mixed up with the one defined in the Vector Function ABI (VFABI) `aarch64_vector_pcs`. The two are kind of related, but given that `aarch64_vector_pcs` is also about SVE, I think we should try to rename the new attribute in a way that avoids mixing it up with the VFABI one. I think we should reserve the `aarch64_vector*` prefix for the VFABI. Also, the `SVE PCS` clearly says that the PCS we want to use is attached to functions that use SVE registers in their interface. We are enforcing the SVE PCS into a function that would not qualify as a function that would have to obey to the rules we want to enforce. 

Here is what I suggest:

The SVE-specific PCS is attached to functions that (lines 878-880)

>  If a subroutine takes
at least one argument in scalable vector registers or scalable predicate
registers, or if it is a function that returns results in such registers,

We could extract this text into the definition of `SVE function`, and rename the attribute introduced in this patch as `aarch64_force_sve_function`. The PCS for SVE would read as:

```
SVE function
^^^^^^^^^^^^

An _SVE function_ is  a subroutine that takes
at least one argument in scalable vector registers or scalable predicate
registers, or a function that returns results in such registers.

Scalable vector registers
^^^^^^^^^^^^^^^^^^^^^^^^^
The Arm 64-bit architecture also defines [...]
z0-z7 are used to pass scalable vector arguments to a subroutine, and to
return scalable vector results from a function. An SVE function (link to the definition) must 
ensure that the entire contents of z8-z23 are preserved across
the call. In other cases [...]

Scalable Predicate Registers
^^^^^^^^^^^^^^^^^^^^^^^^^^^^

[similar changes]

aarch64_force_sve_function
^^^^^^^^^^^^^^^^^^^^^^^^^^

The attribute `aarch64_force_sve_function` can be specified to ensure a subroutine is interpreted as an SVE function (link to definition) even if it does not match the definition of it.  [etc etc.. ].
```

Thanks,

Francesco